src/Instantiator/PropertyInstantiator.php

```diff
@@ -8,8 +8,12 @@
 
 class PropertyInstantiator implements InstantiatorInterface
 {
-    public function instantiate(string $class, array $data)
+    public function instantiate(string $class, ?array $data)
```

Comment:
Shouldn't this have a return type?

Reply:
I guess we can have the `InstantiatorInterface` with a signature like this:

```
public function instantiate(string $class, ?array $data) : \stdClass;
```

Since the type is very generic, I think the correct way to do this would be with a generic type feature which is one that PHP don't have yet. In my humble opinion we should continue without return type for now.

Generic types may be introduced soon in PHP 8.0:

 - https://wiki.php.net/rfc/generics
 - https://github.com/PHPGenerics/php-generics-rfc

Reply:
This change would also cause a breaking change for all the projects that utilize this library. We should be careful with BCs.